File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        dict = defaultdict(int)
        ans=0
        nums=sorted(nums)
        for i in nums:
            if i-1 in dict:
                dict[i]=dict[i-1]+1
                ans = max(ans,dict[i])
            else:
                dict[i]=1
                ans=max(ans,dict[i])
        return ans

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n=heights.size();
        int ans=0;
        int i=0,j=n-1;
        while(i<j){
            if(heights[i]<heights[j]){
                ans=max(ans,heights[i]*(j-i));
                i++;
            }
            else if(heights[i]>heights[j]){
                ans=max(ans,heights[j]*(j-i));
                j--;
            }
            else{
                ans=max(ans,heights[i]*(j-i));
                i++;
            }
        }
        return ans;
    }
};
